### PR TITLE
Little perturbation of diagonal should be applied to the reference of the diagonal values

### PR DESCRIPTION
**Pull request by kyungjoo-kim**
This fixes SPARC problems. trilinos/Trilinos#2714

### CODE REVIEW

**Comment by crtrott** [comment]
Please run spot check when putting in a pull request and post results here.

File: src/batched/KokkosBatched_LU_Team_Internal.hpp [hunk 1]
```diff
@@ -62,11 +59,16 @@ namespace KokkosBatched {
 
         if (tiny != 0) {
           if (member.team_rank() == 0) {
```
Comment (mhoemmen) [inline src/batched/KokkosBatched_LU_Team_Internal.hpp]: For the `tiny != 0` case, the team's Thread 0 modifies the diagonal entry.  Is that access coherent with respect to the rest of the team?  The code that follows needs the (modified) diagonal entry on all threads in the team, not just Thread 0.

Comment (kyungjoo-kim) [reply src/batched/KokkosBatched_LU_Team_Internal.hpp]: the root thread took a reference of the diagonal value and modified it.

Comment (mhoemmen) [reply src/batched/KokkosBatched_LU_Team_Internal.hpp]: @kyungjoo-kim Right, but then the other threads need to see that updated diagonal value.  Are write-to-read memory accesses across different threads in the same team always coherent?

Comment (kyungjoo-kim) [reply src/batched/KokkosBatched_LU_Team_Internal.hpp]: Probably not. I will move the barrier upwards before each thread greb the diagonal values.

Comment (kyungjoo-kim) [reply src/batched/KokkosBatched_LU_Team_Internal.hpp]: @mhoemmen thanks for pointing it out (it is quite subtle).

Comment (crtrott) [reply src/batched/KokkosBatched_LU_Team_Internal.hpp]: Jup that barrier needs moving up

Comment (mhoemmen) [reply src/batched/KokkosBatched_LU_Team_Internal.hpp]: @kyungjoo-kim Cool; I'll be happy to review again when ready :-D .

**Comment by kyungjoo-kim** [comment]
@crtrott here is spot check output.

```
[kyukim @apollo] spotcheck > ../../../master/scripts/test_all_sandia --kokkoskernels-path=/home/kyukim/Work/lib/kokkoskernels/master  --kokkos-path=/home/kyukim/Work/lib/kokkos/master --spot-check cuda intel gcc
Running on machine: apollo
Going to test compilers:  cuda/9.1 intel/16.0.1 gcc/4.8.4 gcc/5.3.0
Testing compiler cuda/9.1
Testing compiler intel/16.0.1
  Starting job cuda-9.1-Cuda_OpenMP-release
  PASSED cuda-9.1-Cuda_OpenMP-release
Testing compiler gcc/4.8.4
  Starting job intel-16.0.1-OpenMP-release
  PASSED intel-16.0.1-OpenMP-release
  Starting job gcc-4.8.4-OpenMP-release
  PASSED gcc-4.8.4-OpenMP-release
Testing compiler gcc/5.3.0
  Starting job gcc-4.8.4-Pthread-release
  PASSED gcc-4.8.4-Pthread-release
  Starting job gcc-5.3.0-Serial-release
  PASSED gcc-5.3.0-Serial-release
#######################################################
PASSED TESTS
#######################################################
cuda-9.1-Cuda_OpenMP-release build_time=796 run_time=2883
gcc-4.8.4-OpenMP-release build_time=138 run_time=317
gcc-4.8.4-Pthread-release build_time=125 run_time=914
gcc-5.3.0-Serial-release build_time=157 run_time=790
intel-16.0.1-OpenMP-release build_time=352 run_time=305
#######################################################
FAILED TESTS
#######################################################
```

**Pull request merged**